grafana-plugin/src/pages/schedule/Schedule.helpers.ts

```diff
@@ -18,6 +19,30 @@ export const getDateTime = (date: string) => {
   return dayjs(date);
 };
 
+export const getUTCByDay = (dayOptions: SelectOption[], by_day: string[], moment: dayjs.Dayjs) => {
+  if (by_day.length > 0 && moment.day() !== moment.utc().day()) {
+    // when converting to UTC, shift starts on a different day,
+    // so we need to update the by_day list
+    // depending on the UTC side, move one day before or after
+    let UTCDays = [];
+    let byDayOptions = [];
+    Object.entries(dayOptions).forEach(([, option]) => byDayOptions.push(option.value));
```

Comment:
By looking at the `dayOptions` type I see they're just an array (`SelectOption[]`), nothing fancy, therefore I don't think you need `Object.entries`, you could get away just by simply iterating

`dayOptions.forEach(({ value }) => byDayOptions.push(value))`